lib/teleterm/clusters/cluster_access_requests.go

```diff
@@ -29,9 +30,60 @@ import (
 	"github.com/gravitational/teleport/lib/teleterm/api/uri"
 )
 
+type ResourceDetails struct {
+	Hostname string
+}
+
 type AccessRequest struct {
 	URI uri.ResourceURI
 	types.AccessRequest
+	ResourceDetails map[string]ResourceDetails
+}
+
+// GetAccessRequest returns a specific access request by ID and includes resource details
+func (c *Cluster) GetAccessRequest(ctx context.Context, req types.AccessRequestFilter) (*AccessRequest, error) {
```

Comment:
Before, `clusters.Cluster` only had one method for `GetAccessRequests` and we let the handler only return the first value when searching for a single request. Now that we need to search for details on a single request and _not_ when listing all, I've added a new method to keep `GetAccessRequests` from making a dial to the auth server and unnecessary logic.